pkgs/servers/sql/sqld/default.nix

```diff
@@ -1,35 +1,45 @@
-{ lib
-, rustPlatform
-, fetchFromGitHub
-, pkg-config
-, protobuf
-, openssl
-, sqlite
-, zstd
-, stdenv
-, darwin
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  protobuf,
+  openssl,
+  sqlite,
+  zstd,
+  stdenv,
+  darwin,
+  cmake,
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "sqld";
-  version = "0.17.2";
+  version = "0.24.18";
 
   src = fetchFromGitHub {
-    owner = "libsql";
-    repo = "sqld";
-    rev = "v${version}";
-    hash = "sha256-KoEscrzkFJnxxJKL/2r4cY0oLpKdQMjFR3daryzrVKQ=";
+    owner = "tursodatabase";
+    repo = "libsql";
+    rev = "libsql-server-v${version}";
+    hash = "sha256-/0Sk55GBjUk/FeIoq/hGVaNGB0EM8CxygAXZ+ufvWKg=";
   };
 
+  cargoBuildFlags = [
+    "--bin"
+    "sqld"
+  ];
+
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = "${src}/Cargo.lock";
     outputHashes = {
-      "libsqlite3-sys-0.26.0" = "sha256-JzSGpqYtkIq0mVYD0kERIB6rmZUttqkCGne+M4vqTJU=";
-      "octopod-0.1.0" = "sha256-V16fOlIp9BCpyzgh1Aei3Mra/y15v8dQFA8tHdOwZm4=";
+      "console-api-0.5.0" = "sha256-MfaxtzOqyblk6aTMqJGRP+123aK0Kq7ODNp/3lehkpQ=";
+      "hyper-rustls-0.24.1" = "sha256-dYN42bnbY+4+etmimrnoyzmrKvCZ05fYr1qLQFvzRTk=";
+      "rheaper-0.2.0" = "sha256-u5z6J1nmUbIQjDDqqdkT0axNBOvwbFBYghYW/r1rDHc=";
+      "s3s-0.10.1-dev" = "sha256-y4DZnRsQzRNsCIp6vrppZkfXSP50LCHWYrKRoIHYPik=";
     };
   };
 
   nativeBuildInputs = [
+    cmake
```

Comment:
What's this for?  This usually indicates a vendored C/C++ library is being built, which we usually don't want (it's better to get the library from Nixpkgs).

Reply:
It's the only way I could get it to build. I'm probably out of my depth here then.

Reply:
Sadly it looks like libsql-ffi doesn't support building any other way, so I guess we can do this.